### PR TITLE
HDDS-4107. replace scmID with clusterID for container and volume at Datanode side.

### PR DESCRIPTION
**Pull request by mukul1987**
## What changes were proposed in this pull request?

Use clusterID for initializing the cluster

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-4107

## How was this patch tested?
Added new tests

### CODE REVIEW

**Review by bharatviswa504 — COMMENTED** [review]
Overall LGTM.
Few minor comments/questions.

File: hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/ScmTestMock.java [hunk 1]
```diff
@@ -357,4 +357,12 @@ public void setScmId(String id) {
   public String getScmId() {
     return scmId;
   }
+
+  public String getClusterId() {
```
[the remark this answers is not in the document]
Comment (mukul1987) [reply hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/ScmTestMock.java:361]: Done

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/HddsVolume.java [hunk 2]
```diff
@@ -52,9 +52,9 @@
  * <p>
  * The disk layout per volume is as follows:
  * <p>../hdds/VERSION
- * <p>{@literal ../hdds/<<scmUuid>>/current/<<containerDir>>/<<containerID
+ * <p>{@literal ../hdds/<<clusterUuid>>/current/<<containerDir>>/<<containerID
  * >>/metadata}
```
[the remark this answers is not in the document]
Comment (mukul1987) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/HddsVolume.java:56]: done

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/utils/HddsVolumeUtil.java [hunk 3]
```diff
@@ -194,26 +195,32 @@ public static boolean checkVolume(HddsVolume hddsVolume, String scmId, String
...
+      if (!clusterDir.mkdir()) {
+        logger.error("Unable to create scmDir {}", clusterDir);
         return false;
       }
       return true;
     } else if(hddsFiles.length == 2) {
-      // The files should be Version and SCM directory
       if (scmDir.exists()) {
+        logger.error("Volume {} is in Inconsistent state, and contains the" +
+            "SCM Directory:{} which is in previous format, please upgrade" +
```
[the remark this answers is not in the document]
Comment (mukul1987) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/utils/HddsVolumeUtil.java]: done

**Review by bharatviswa504 — APPROVED** [review]
+1 LGTM.

We can revisit the upgrade test once we have the utilities to rename directory and container yaml files.

**Pull request merged**